### PR TITLE
http: fix encoding of IPv4 peers with 16-byte addresses

### PR DESCRIPTION
**Pull request by mrd0ll4r**
This is #189 for the refactor branch.

### CODE REVIEW

**Comment by jzelinskie** [comment]
I'd rather have an extra test for storage implementations than running the risk of mistakenly reformatted IPv4-compatible IPv6 addresses into IPv4 ones.

**Comment by mrd0ll4r** [comment]
What about this: Let's introduce a `compact4` and a `compact6` function, to be called for `peers4` and `peers6` respectively. The former just calls `To4` on every IP, the latter calls `To16` ?

**Comment by jzelinskie** [comment]
That sounds fair.

**Comment by mrd0ll4r** [comment]
Alright, now using two functions, can you take another look please?

File: frontend/http/writer.go [hunk 1]
```diff
@@ -82,8 +82,15 @@ func WriteScrapeResponse(w http.ResponseWriter, resp *bittorrent.ScrapeResponse)
 	})
 }
 
-func compact(peer bittorrent.Peer) (buf []byte) {
-	buf = []byte(peer.IP)
+func compact4(peer bittorrent.Peer) (buf []byte) {
+	buf = []byte(peer.IP.To4())
```
Comment (jzelinskie) [inline frontend/http/writer.go]: In the off chance that peer.IP.To4() returns nil, this isn't going to fail, but will provide an invalid response.
Let's panic if it's nil.

Comment (mrd0ll4r) [inline frontend/http/writer.go]: yeah I thought about that too, but was hesitant to panic. Still, added it now.

**Comment by jzelinskie** [comment]
LGTM

**Pull request merged**